src/plugins/embeddable/public/lib/filterable_embeddable/should_fetch.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fastIsEqual from 'fast-deep-equal';
+import { Observable } from 'rxjs';
+import { map, distinctUntilChanged, skip, startWith } from 'rxjs/operators';
+import type { Filter, Query, TimeRange } from '@kbn/es-query';
+import { COMPARE_ALL_OPTIONS, onlyDisabledFiltersChanged } from '@kbn/es-query';
+import { EmbeddableInput } from '../embeddables';
+
+export const shouldRefreshFilterCompareOptions = {
+  ...COMPARE_ALL_OPTIONS,
+  // do not compare $state to avoid refreshing when filter is pinned/unpinned (which does not impact results)
+  state: false,
+};
+
+type FilterableEmbeddableInput = EmbeddableInput & {
+  filters?: Filter[];
+  query?: Query;
+  timeRange?: TimeRange;
+  timeslice?: [number, number];
+};
+
+export function shouldFetch$<
+  TFilterableEmbeddableInput extends FilterableEmbeddableInput = FilterableEmbeddableInput
+>(
+  updated$: Observable<unknown>,
+  getInput: () => TFilterableEmbeddableInput
+): Observable<TFilterableEmbeddableInput> {
+  return updated$.pipe(map(() => getInput())).pipe(
```

Comment:
Just out of curiousity, why take an observable of type unknown and a getInput function here instead of taking an observable of type `TFilterableEmbeddableInput`? Is this so you can `startWith` `getInput`?

Reply:
> Is this so you can startWith getInput?

Yes. I want to preload distinctUntilChanged so next observable firing only passes distinctUntilChanged if `a` is different than `b`, not because `b` is undefined

Reply:
Could you handle that from within the `distinctUntilChanged` callback where if one is `undefined` you `return true`? Or it could be handled with a `skipWhile`?

I see that you're also using the `getInput` callback to filter out the `controlledBy` in maps, so it's probably simpler to leave it this way.

Reply:
> I see that you're also using the getInput callback to filter out the controlledBy in maps, so it's probably simpler to leave it this way.

And its being used to filter out searchSessionId in the unique case where maps has "filter by map bounds" enabled. When "filter by map bounds" is enabled, moving the map causes the dashboard to re-filter all panels by new map bounds. This refiltering changes the searchSessionId. In this case, the map panel should not refetch data so it ignores searchSessionId.

Reply:
That's true. It's a bit more idiomatic for RXJS to have an observable of the type & pass in a different observable in the few cases that something non-default is required, but I think both solutions are equally readable -  which is the most important - so I wouldn't suggest a refactor.